sources/simulators.swift

```diff
@@ -1,12 +1,36 @@
 import Foundation
 
-enum SimulatorFetchError: String, Error {
-    case simctlFailed = "Running `simctl list` failed"
-    case failedToReadOutput = "Failed to read output from simctl"
-    case noBootedSimulators = "No simulators are currently booted"
+struct Simulator {
+    fileprivate enum State: String {
+        case shutdown = "Shutdown"
+        case booted = "Booted"
+    }
+
+    let name: String
+    let udid: String
+
+    fileprivate init?(dictionary: [String: String]) {
+        guard let state = dictionary["state"].map(State.init), let udid = dictionary["udid"],
+            let name = dictionary["name"], state == .booted else
+        {
+            return nil
+        }
+
+        self.name = name
+        self.udid = udid
+    }
 }
 
-func getBootedSimulators() throws -> [String] {
+func getSimulators(named name: String, from simulators: [Simulator]) throws -> [Simulator] {
+    let matchingSimulators = simulators.filter { $0.name.lowercased() == name.lowercased() }
```

Comment:
Just a style difference but maybe:
```
guard let matchingSimulators = simulators.filter { $0.name.lowercased() == name.lowercased() },
    !matchingSimulators.isEmpty else {
        throw SimulatorFetchError.noMatchingSimulators(name: name)
}
```

Reply:
I generally prefer the ifs here since I think that's clearer to read the positive conditions for these cases and since we don't need anything to be unwrapped below, although I do see that our style guidelines got a little looser around that recently. Same with the one below

Reply:
Kk thanks for the clarification =)

Reply:
Yeah the new `guard` guidance is great :+1: